sdk/cosmos/azure-cosmos/src/main/java/com/azure/cosmos/implementation/ClientSideRequestStatistics.java

```diff
@@ -65,6 +65,25 @@ public ClientSideRequestStatistics(DiagnosticsClientContext diagnosticsClientCon
         this.globalEndpointManager = globalEndpointManager;
     }
 
+    public ClientSideRequestStatistics(ClientSideRequestStatistics toBeCloned) {
+        this.diagnosticsClientContext = toBeCloned.diagnosticsClientContext;
```

Comment:
does this needs to be deep clone? like the retryContext

Reply:
Not possible - DiagnosticsClientContext is an interface implemented for example by RxDocumentClientImpl - passed around as reference everywhere and as such is inherently thread-safe